src/main/java/net/bhardy/bizzo/billing/CycleType.java

```diff
@@ -15,19 +15,32 @@
  */
 package net.bhardy.bizzo.billing;
 
+import java.time.Period;
+
 /**
  * Billing cycles typically fall roughly into the following time cycles.
  * <p>
  * You can use PolicyFilters to make behaviour more specific.
  */
 public enum CycleType {
-    DAILY,
-    WEEKLY,
-    BIWEEKLY,
-    SEMIMONTHLY,
-    MONTHLY,
-    BIMONTHLY,
-    QUARTERLY,
-    SEMIANNUALLY,
-    ANNUALLY
+
+    DAILY(Period.ofDays(1)),
+    WEEKLY(Period.ofWeeks(1)),
+    BIWEEKLY(Period.ofWeeks(2)),
+    //SEMIMONTHLY(Period.ofDays(1)),
+    MONTHLY(Period.ofMonths(1)),
+    BIMONTHLY(Period.ofMonths(2)),
+    QUARTERLY(Period.ofMonths(3)),
+    SEMIANNUALLY(Period.ofMonths(6)),
+    ANNUALLY(Period.ofYears(1));
+
+    private final Period period;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this value-based field transient so it is not included in the serialization of this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3437)